frontend/public/module/k8s/k8s-models.ts

```diff
@@ -72,3 +80,57 @@ export const modelFor = (ref: K8sResourceKindReference) => {
  * NOTE: This will not work for CRDs defined at runtime, use `connectToModels` instead.
  */
 export const allModels = () => k8sModels;
+
+/**
+ * Use this hook to find the model for resources using only group and plural/kind
+ */
+export const useModelFinder = () => {
+  const referenceForGroupVersionPlural = (group: string) => (version: string) => (plural: string) =>
+    [group, version, plural].join('~');
+
+  const models: ImmutableMap<string, K8sKind> = useSelector(({ k8s }) =>
+    k8s.getIn(['RESOURCES', 'models']),
+  );
+  const pluralsToModelMap = models.reduce((acc, curr) => {
+    const ref = referenceForGroupVersionPlural(curr.apiGroup)(curr.apiVersion)(curr.plural);
+    acc[ref] = curr;
+    return acc;
+  }, {});
+  const groupVersionMap: DiscoveryResources['groupVersionMap'] = useSelector(({ k8s }) =>
+    k8s.getIn(['RESOURCES', 'groupToVersionMap']),
+  );
+
+  const findModel = (group: string, resource: string) => {
+    if (!group) {
+      const refPlural = referenceForGroupVersionPlural(group)('v1')(resource);
+      const refKind = referenceForGroupVersionKind('core')('v1')(resource);
```

Comment:
We shouldn't be calling `referenceForGroupVersionKind` in this method since we should never have the kind. (Let's open bugs for cluster operators that have incorrect `relatedObjects` stanzas.)